megamek/src/megamek/common/verifier/TestSupportVehicle.java

```diff
@@ -14,34 +14,602 @@
  */
 
 package megamek.common.verifier;
-import megamek.common.Tank;
+import megamek.common.*;
+import megamek.common.annotations.Nullable;
+import megamek.common.logging.DefaultMmLogger;
 import megamek.common.util.StringUtil;
+import megamek.common.weapons.flamers.VehicleFlamerWeapon;
+import megamek.common.weapons.lasers.CLChemicalLaserWeapon;
+
+import java.math.BigInteger;
+import java.util.*;
 
 /**
  * Author: arlith
  */
-public class TestSupportVehicle extends TestTank {
-
-    /**
-     * Gives the weight of a single point of armor at a particular BAR for a 
-     * given tech level.
-     */
-    public static final double[][] SV_ARMOR_WEIGHT = 
-        {{0.0, 0.0, 0.0, 0.0, 0.0, 0.0},
-         {0.0, 0.0, 0.0, 0.0, 0.0, 0.0},
-         {.040, .025, .016, .013, .012, .011},
-         {.060, .038, .024, .019, .017, .016},
-         {.000, .050, .032, .026, .023, .021},
-         {.000, .063, .040, .032, .028, .026},
-         {.000, .000, .048, .038, .034, .032},
-         {.000, .000, .056, .045, .040, .037},
-         {.000, .000, .000, .051, .045, .042},
-         {.000, .000, .000, .057, .051, .047},
-         {.000, .000, .000, .063, .056, .052},};
+public class TestSupportVehicle extends TestEntity {
+
+    /**
+     * Support vehicle categories for construction purposes. Most of these match with a particular movement
+     * mode, but the construction rules treat naval and rail units as single types.
+     */
+    public enum SVType implements ITechnologyDelegator {
+        AIRSHIP (300, EntityMovementMode.AIRSHIP,
```

Comment:
I'm incapable of figuring out what the values in these data structures mean, is it possible to put an explanatory comment?